### ISSUE TITLE
More accurate time tracker for Agony

### ISSUE THREAD

**Issue by scbdgina** [Lleafll/doom-shards#3]
Is there a way to update the tracking shard regen for Agony?  As we know now it is not a chance per tick to regen a shard but a smaller boost to the natural shard regen for Affliction locks. I don't have the math behind it on memory but i know the formula is out there that shows how much quicker shards regen based on number of targets with agony.

**Comment by Lleafll** [comment]
I agree that the current implementation is hardly useful. I have already looked into options but there are several issues at play here:

- As you are certainly aware, accumulator after application and shard gain is random. Additionally accumulator increase is random and therefore rollover after shard gain is also random. This means in comparison to Doom any attempt at shard prediction for Agony is highly inaccurate.
- I could not find a closed formulation for calculating the new shard gain chance for the upcoming tick based on the previous tick. This means I would need to calculate the chance based on all previous ticks (modified by target count at the time) since the last shard gain. Which is rather inefficient and not worth the cost imo. I have solved this numerically and pre-computed values based on the average target count of previous ticks but have not further implemented this approach due to the concerns raised in the previous and following points.
- Shard gain is too frequent. Any predictions shown will have changed two GCDs later. The time window for gaining shards is often in the range of a few seconds. So I personally doubt the gameplay benefit of Agony shard prediction.

That said I guess the best implementation would be to show a new accumulator over the sum of ticks according to
`sum( 1 / sqrt(targets_for_tick) )`

Another approach conforming with your suggestion would be to only show the time until shard-cap. That would eliminate a lot of "business" in the UI and still give a guideline for shard usage.

I'll have to implement both options and see how they play out but please feel free to reply with your thoughts.
